lib/mysql/mysql.go

```diff
@@ -612,31 +614,63 @@ func (c *Connection) decodePacket(body []byte) (PacketInfo, error) {
 	}
 }
 
+// if n, _ := io.Read(body), returns a hex representation of body[:n] that is at most 96 bytes long.
+func trunc(body []byte, n int) string {
+	if body == nil {
+		return "<nil>"
+	}
+	if n > len(body) {
+		n = len(body)
+	}
+	if n < 1 {
+		return "<empty>"
+	}
+	if n < 48 {
+		return fmt.Sprintf("%x", body[:n])
+	}
+	return fmt.Sprintf("%x...[%d bytes]...%x", body[:16], n - 32, body[n-16:])
+}
+
 // Read a packet and sequence identifier off of the given connection
 func (c *Connection) readPacket() (*ConnectionLogEntry, error) {
-	// @TODO @FIXME Find/use conventional buffered packet-reading functions, handle timeouts / connection reset / etc
 	reader := bufio.NewReader(c.Connection)
 	var header [4]byte
-	n, err := reader.Read(header[:])
+	n, err := io.ReadFull(reader, header[:])
 	if err != nil {
-		return nil, fmt.Errorf("Error reading packet header: %s", err)
+		return nil, fmt.Errorf("error reading packet header: %s", err)
 	}
 	if n != 4 {
-		return nil, fmt.Errorf("Wrong number of bytes returned (got %d, expected 4)", n)
+		return nil, fmt.Errorf("wrong number of bytes returned (got %d, expected 4)", n)
```

Comment:
nice

Reply:
Strictly, this shouldn't be reachable, since `io.ReadFull` should return a non-`nil` `err` if it's unable to read the entire header, but it doesn't hurt.